### PR TITLE
build-script: use more terse checks

### PR DESCRIPTION
**Pull request by compnerd**
This uses the bash extension of regex based comparison rather than the
more portable shell approach which is a bit more terse.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @Rostepher @edymtt @xiaobai

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: utils/build-script-impl [hunk 1]
```diff
@@ -854,34 +854,19 @@ if [[ "${SKIP_RECONFIGURE}" ]]; then
...
-case "${INSTALLABLE_PACKAGE}" in
-    /*) ;;
-    "") ;;
-    *)
-        echo "installable-package must be an absolute path (was '${INSTALLABLE_PACKAGE}')"
-        exit 1
-        ;;
-esac
+# WORKSPACE, BUILD_DIR, and INSTALLABLE_PACKAGE must be absolute paths
+if ! [[ ${WORKSPACE} =~ ^/ ]] ; then
```
Comment (Rostepher) [inline utils/build-script-impl]: Should we be quoting these variable expansions?

Comment (compnerd) [reply utils/build-script-impl]: No, it shouldn't be needed here.  The quoting behaviour of bash is far saner than that of POSIX sh.

**Pull request merged**